UM/Settings/ContainerRegistry.py

```diff
@@ -441,12 +442,10 @@ def removeContainer(self, container_id: str) -> None:
         if container_id in self.metadata:
             if container is None:
                 # We're in a bit of a weird state now. We want to notify the rest of the code that the container
-                # has been deleted, but due to lazy loading, it hasnt even been loaded yet. The issues is that in order
-                # to notify the rest of the code, we need to actually *have* the container. So we need to load it
-                # in order to remove it...
-                provider = self.source_provider.get(container_id)
-                if provider:
-                    container = provider.loadContainer(container_id)
+                # has been deleted, but due to lazy loading, it hasn't been loaded yet. The issues is that in order
+                # to notify the rest of the code, we need to actually *have* the container. So an empty instance 
+                # container is created, which is emitted with the containerRemoved signal and contains the metadata
+                container = EmptyInstanceContainer(container_id)
```

Comment:
The metadata of the old container should be transferred to the new EmptyInstanceContainer, because some functions that are activated by the containerRemoved signal take actions based on the container's metadata (e.g. see CloutOutputDeviceManager.py:456 or IntentcategoryModel:95).

Reply:
Looked through our own codebase and imo the branches that are executed with a request to non-existing metadata, such as: `type` don't break when this key isn't present. That being said It could be that an external plugin uses the metadata. So just to be safe I created a commit in which the metadata is copied to the EmptyInstanceContainer.

Btw, the CloudOutputDeviceManager.py:456 will never request the metadata because it will first check if its a `GlobalStack` instance which it is not.